wp-content/plugins/tapestry/controller/class.user-controller.php

```diff
@@ -0,0 +1,120 @@
+<?php
+/**
+ * Add/update/retrieve User progress
+ * 
+ */
+class TapestryUserController {
+
```

Comment:
For errors, could you please follow what Andrew is doing in PR #20 as I find it a bit more elegant? So add in an array for error types like ```    const ERRORS = [
        'INVALID_POST_ID' => [
            'MESSAGE' => 'PostID is invalid',
            'STATUS' => ['status' => 404]
        ]
    ];```
and add in your own errors in here. Then also copy in the `throwsError` function in here and use that instead of the generic `throw new Exception`.

Reply:
That sounds good. Since him and I may have the same errors, should we create a new issue to merge this into a class to avoid duplication?